src/Numerics/ODESolvers/AdditiveRungeKuttaMethod.jl

```diff
@@ -139,49 +144,64 @@ mutable struct AdditiveRungeKutta{
 
         # The code throughout assumes SDIRK implicit tableau so we assert that
         # here.
-        for is in 2:Nstages
-            @assert RKA_implicit[is, is] ≈ RKA_implicit[2, 2]
-        end
+        # for is in 2:Nstages
+        #     @assert RKA_implicit[is, is] ≈ RKA_implicit[2, 2]
+        # end
+
+        # NOTE: this is only for composing an ARK method with
+        # a MIS timestepper
+        # TODO: Clean this up
+
+        implicit_solvers = Dict()
+
+        rk_diag = unique(diag(RKA_implicit))
+        # Remove all zero entries from `rk_diag`
+        # so we build all unique implicit solvers (parameterized by the
+        # corresponding RK coefficient)
+        filter!(c -> c !== 0, rk_diag)
```

Comment:
```suggestion
        filter!(c -> c != 0, rk_diag)
```

You want this to be `!=` since:

```julia
julia> 0.0 !== 0
true

julia> 0.0 != 0
false
```

Reply:
Oh my goodness!! This was a nasty bug!! Thanks @jkozdon !!